src/high-level/types/single-float.lisp

```diff
@@ -22,6 +22,37 @@
   matrix/single-float
   vector/single-float)
 
+;; Might want to inline this
+(defun %scalar= (s1 s2 epsilon)
+  "For equality checks of inexact scalars."
+  (declare (type number s1 s2) ; you can use this on complex but it might be slower because of the sqrt and two multiplies. Maybe.
+           (type real epsilon))
+  (<= (abs (- s1
+              s2))
+      epsilon))
+
+(defmethod =-lisp ((scalar1 rational) (scalar2 rational) &optional epsilon)
+  (declare (ignore epsilon))
+   "Rationals (integers and ratios) should be compared exactly."
+  (common-lisp:= scalar1 scalar2))
+
```

Comment:
Could we move this to a separate file like `numerical-equality.lisp`? The extensible function defined in `abstract-tensor` should probably be moved there too. That way we can knock out a bunch of general/generic stuff all in one go, and we don't have it hidden here in `single-float.lisp`